torch/serialization.py

```diff
@@ -1112,6 +1116,22 @@ def load_tensor(dtype, numel, key, location):
         name = f'data/{key}'
 
         storage = zip_file.get_storage_from_record(name, numel, torch.UntypedStorage)._typed_storage()._untyped_storage
+        # check if byteswapping is needed and swap here if it is needed
+        byteencodingname = 'byteencoding'
+        if zip_file.has_record(byteencodingname):
+            encodingdata = zip_file.get_record(byteencodingname)
+            need_byteswap = True
+            elem_size = torch._utils._element_size(dtype)
+            if elem_size == 1:
+                need_byteswap = False
+            if (encodingdata == b'le' and sys.byteorder == 'little') or (encodingdata == b'be' and sys.byteorder == 'big'):
```

Comment:
Should probably throw an error if `encodingdata` is some other unexpected value, just to be safe

Reply:
Thanks, will do. Any specific error type you could recommend using here?

Reply:
I've used ValueError here. Please let me know if another error type should be used.